api/src/services/import-export.ts

```diff
@@ -132,29 +130,20 @@ export class ImportService {
 				return await service.upsertOne(value, { bypassEmitAction: (action) => nestedActionEvents.push(action) });
 			});
 
+			const transform = (value: any, _field: string) => {
+				if (value === 'NULL') return null;
```

Comment:
Inserting an empty string check here to return `null` would fix the empty field issue but introduce a problem when you _do_ actually want to input an empty string because the field is non-nullable. Careful there.

Reply:
Yeh interesting problem. Am actually curious as to why the empty string value in your example wasn't interpreted as `'NULL'` by papaparse as that is what I have experienced.

We can add empty string check = null and then wait for the first issue regarding it to pop up and address again.

Reply:
> We can add empty string check = null and then wait for the first issue regarding it to pop up and address again.

I'd rather not do that. That's the type of tech debt that I'd like to prevent.

Reply:
> Am actually curious as to why the empty string value in your example wasn't interpreted as 'NULL'

Just took a quick search over in the papaparse repo and found the unit test confirming my experience:

https://github.com/mholt/PapaParse/blob/841e1d420dd2758cccb60a2bc1f2fcffd327c251/tests/test-cases.js#L327-L334

Also sequential empty fields are all expected to be empty strings:

https://github.com/mholt/PapaParse/blob/841e1d420dd2758cccb60a2bc1f2fcffd327c251/tests/test-cases.js#L370-L377

Reply:
We need to either submit a PR to `PapaParse` allowing to make that configurable or revert back to `csv-parser` and submit a PR there allowing to auto-detect delimiters (or implement on our side).

Reply:
There are two sides to every problem. Either the way it worked before was just lucky it did, but might be wrong according to what one might expect, eg. an empty string should just be an empty string.
Or all undefined, null and empty strings should be null.
In either case, we can update the PR to reflect the side we choose. But if you think about it, even papaparse won't be able to tell the difference. It is the same as `'NULL'`; Is it just a string containing the word "NULL" or is it because papaparse parsed it as a `null` value.

It becomes hard to tell, but better to then make a decision, and document it clearly.

Reply:
By reusing the same transform logic previously used for `csv-parser`, it now behaves exactly the same as before 👍 
https://github.com/directus/directus/blob/ca51a2dbc14e8d33028d1177f336ae529eec4522/api/src/services/import-export/index.ts#L132-L146